src/main/scala/tech/sourced/gemini/ConnectedComponents.scala

```diff
@@ -0,0 +1,125 @@
+package tech.sourced.gemini
+
+import java.nio.ByteBuffer
+
+import com.datastax.driver.core.{Session, SimpleStatement}
+import org.slf4j.{Logger => Slf4jLogger}
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ListBuffer
+
+case class FileHash(sha1: String, value: ByteBuffer)
+
+class DBConnectedComponents(
+                              log: Slf4jLogger,
+                              conn: Session,
+                              keyspace: String = Gemini.defautKeyspace) extends ConnectedComponents(log) {
+  def getHashtables(): List[Byte] = {
+    val hashtablesCql = s"SELECT DISTINCT hashtable FROM $keyspace.hashtables"
+
+    conn
+      .execute(new SimpleStatement(hashtablesCql))
+      .asScala
+      .map(_.getByte("hashtable"))
+      .toList.
+      sorted
+  }
+
+  def getHashValues(hashtable: Byte): Iterable[FileHash] = {
+    val cql = s"SELECT sha1, value FROM $keyspace.hashtables WHERE hashtable=$hashtable"
```

Comment:
Do you think it would be possible for `hashtables` table, to follow the convention we use for [meta table](https://github.com/src-d/gemini/blob/master/src/main/scala/tech/sourced/gemini/Gemini.scala?utf8=%E2%9C%93#L180)?

Reply:
sorry, I don't really get the idea behind meta table. But I added table name as a parameter.